components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -254,10 +268,19 @@ void RRTMGPRadiation::run_impl (const Real dt) {
   gas_concs.init(gas_names_1d,m_ncol,m_nlay);
   auto tmp2d = m_buffer.tmp2d;
   for (int igas = 1; igas <= m_ngas; igas++) {
-    parallel_for(Bounds<2>(m_ncol,m_nlay), YAKL_LAMBDA(int icol, int ilay) {
-        tmp2d(icol,ilay) = gas_vmr(icol,ilay,igas);
-    });
-    gas_concs.set_vmr(gas_names_1d(igas), tmp2d);
+    auto name = gas_names_1d(igas);
+    if (name=="h2o") {
+      auto d_temp  = m_rrtmgp_fields_in.at("qv").get_reshaped_view<const Real**>();
+      parallel_for(Bounds<2>(m_ncol,m_nlay), YAKL_LAMBDA(int icol, int ilay) {
```

Comment:
Agreed, I'm not sure if we can change it though.  This chunk of code was already in master, here it has just been copied twice to facilitate the `if` statement.  @brhillman or @tcclevenger  may be able to comment on whether or not we can shift to a Kokkos loop here.

Reply:
Why can't it be changed?

Reply:
I don't know, sense this chunk of code is @brhillman 's originally I wanted to give him a chance to comment before I change it.

Reply:
I agree that things in the AD interface should use kokkos parallel syntaxes. Before @tcclevenger made the fixes for packs though, everything in this layer was using YAKL, just because it made life a little easier in that we were doing all the wrapping with YAKL arrays. So I'm perfectly happy having this all changed over to kokkos. The `gas_names` shouldn't be used by RRTMGP, it's all just temporary for populating the `gas_concs` object, so that can really be done however it makes the most sense to. Once that object is populated (with gas names and gas volume mixing ratios), that's all that RRTMGP cares about. Order doesn't matter.